### DISCUSSION TITLE
Catit Pixi smart fountain mapping scale

### DISCUSSION THREAD

**Discussion by blamaz** [make-all/tuya-local#564] [General]
My only Tuya devices are the Catit Pixi smart fountain and smart feeder. Only the smart fountain is use with this integration.
On the last few updates, I've found out that you've removed the scale from dps id 4 and 5 and now thoses entity show an increase of time (days) since the last initialization using the corresponding boolean switch. I found this to be less usefull than a countdown and both value are the sames and keep increasing without limits. The dps id 4 should be a 30 days countdown and the dps id 5, a 60 days countdown. Also, there is no use for the value of mapping invert: true if there is no scale. Could you add theses mapping again.

secondary_entities:

    entity: sensor
    name: Filter life
    icon: "mdi:air-filter"
    category: diagnostic
    dps:
        id: 4
        type: integer
        name: sensor
        precision: 0
        range:
        min: 0
        max: 43200
        mapping:
            scale: 1440
            step: 1
            invert: true
            unit: d
    entity: sensor
    name: Pump maintenance
    icon: "mdi:pump"
    category: diagnostic
    dps:
        id: 5
        type: integer
        name: sensor
        precision: 0
        range:
        min: 0
        max: 86400
        mapping:
            scale: 1440
            step: 1
            invert: true
            unit: d

Here is the info from Tuya developper tool

{
"result": {
"category": "cwysj",
"functions": [
{
"code": "switch",
"dp_id": 1,
"type": "Boolean",
"values": "{}"
},
{
"code": "water_reset",
"dp_id": 6,
"type": "Boolean",
"values": "{}"
},
{
"code": "filter_reset",
"dp_id": 7,
"type": "Boolean",
"values": "{}"
},
{
"code": "pump_reset",
"dp_id": 8,
"type": "Boolean",
"values": "{}"
},
{
"code": "uv",
"dp_id": 10,
"type": "Boolean",
"values": "{}"
},
{
"code": "uv_runtime",
"dp_id": 11,
"type": "Integer",
"values": "{"unit":"s","min":0,"max":10800,"scale":0,"step":1}"
}
],
"status": [
{
"code": "switch",
"dp_id": 1,
"type": "Boolean",
"values": "{}"
},
{
"code": "water_time",
"dp_id": 3,
"type": "Integer",
"values": "{"unit":"min","min":0,"max":7200,"scale":0,"step":1}"
},
{
"code": "filter_life",
"dp_id": 4,
"type": "Integer",
"values": "{"unit":"min","min":0,"max":43200,"scale":0,"step":1}"
},
{
"code": "pump_time",
"dp_id": 5,
"type": "Integer",
"values": "{"unit":"min","min":0,"max":86400,"scale":0,"step":1}"
},
{
"code": "water_reset",
"dp_id": 6,
"type": "Boolean",
"values": "{}"
},
{
"code": "filter_reset",
"dp_id": 7,
"type": "Boolean",
"values": "{}"
},
{
"code": "pump_reset",
"dp_id": 8,
"type": "Boolean",
"values": "{}"
},
{
"code": "uv",
"dp_id": 10,
"type": "Boolean",
"values": "{}"
},
{
"code": "uv_runtime",
"dp_id": 11,
"type": "Integer",
"values": "{"unit":"s","min":0,"max":10800,"scale":0,"step":1}"
},
{
"code": "water_level",
"dp_id": 12,
"type": "Enum",
"values": "{"range":["level_1","level_2","level_3"]}"
}

**Comment by make-all** [comment]
The range was added back in yesterday after the same request was made elsewhere.
